pkg/ruler/remotequerier_test.go

```diff
@@ -709,27 +709,85 @@ func TestRemoteQuerier_BackoffRetry(t *testing.T) {
 }
 
 func TestRemoteQuerier_StatusErrorResponses(t *testing.T) {
-	mockClientFn := func(ctx context.Context, req *httpgrpc.HTTPRequest, _ ...grpc.CallOption) (*httpgrpc.HTTPResponse, error) {
-		return &httpgrpc.HTTPResponse{
+	var (
+		errorResp = &httpgrpc.HTTPResponse{
 			Code: http.StatusUnprocessableEntity,
 			Headers: []*httpgrpc.Header{
 				{Key: "Content-Type", Values: []string{"application/json"}},
 			},
 			Body: []byte(`{
 				"status": "error","errorType": "execution"
 			}`),
-		}, nil
+		}
+		error4xx = status.Error(http.StatusUnprocessableEntity, "this is a 4xx error")
+		error5xx = status.Error(http.StatusInternalServerError, "this is a 5xx error")
+	)
+	testCases := map[string]struct {
```

Comment:
Add a test case returning a non httpgrpc error (e.g. `errors.New("mocked error")`). We expect the log.